eventstore/memory/iterator.go

```diff
@@ -0,0 +1,27 @@
+package memory
+
+import "github.com/hallgren/eventsourcing/core"
+
+type iterator struct {
+	events   []core.Event
+	position int
+	event    core.Event
```

Comment:
Isn't it redundant to store an extra value of the event in addition to the events and position?
I mean that the Value method could just check the position and return `i.events[i.pos]`?

Reply:
As the i.events is a slice and it's zero indexed I need extra code to handle the initial 0 position when calling the Next method. This was the simplest solution my brain could handle.